src/crh/constraints.rs

```diff
@@ -23,3 +23,22 @@ pub trait FixedLengthCRHGadget<H: FixedLengthCRH, ConstraintF: Field>: Sized {
         input: &[UInt8<ConstraintF>],
     ) -> Result<Self::OutputVar, SynthesisError>;
 }
+
+pub trait TwoToOneCRHGadget<H: TwoToOneCRH, ConstraintF: Field>: Sized {
+    type OutputVar: EqGadget<ConstraintF>
+        + ToBytesGadget<ConstraintF>
+        + CondSelectGadget<ConstraintF>
+        + AllocVar<H::Output, ConstraintF>
+        + R1CSVar<ConstraintF>
+        + Debug
+        + Clone
+        + Sized;
+
+    type ParametersVar: AllocVar<H::Parameters, ConstraintF> + Clone;
+
+    fn evaluate_both(
```

Comment:
`evaluate_both` is a misnomer, no? Why not just `evaluate`?

Reply:
I changed it back to `evaluate`